Tests/SwiftLintFrameworkTests/FunctionBodyLengthRuleTests.swift

```diff
@@ -21,15 +21,15 @@ private func violatingFuncWithBody(_ body: String) -> String {
 class FunctionBodyLengthRuleTests: XCTestCase {
 
     func testFunctionBodyLengths() {
-        let longFunctionBody = funcWithBody(repeatElement("x = 0\n", count: 39).joined())
+        let longFunctionBody = funcWithBody(repeatElement("x = 0\n", count: 38).joined())
         XCTAssertEqual(violations(longFunctionBody), [])
 
         let longerFunctionBody = violatingFuncWithBody(repeatElement("x = 0\n", count: 40).joined())
         XCTAssertEqual(violations(longerFunctionBody), [StyleViolation(
             ruleDescription: FunctionBodyLengthRule.description,
             location: Location(file: nil, line: 1, character: 1),
             reason: "Function body should span 40 lines or less excluding comments and " +
-            "whitespace: currently spans 41 lines")])
+            "whitespace: currently spans 42 lines")])
```

Comment:
Sorry, I fail to see how this was previously incorrect and is now correct.

We're adding 40 lines to the function body additional to its first line:

```swift
violatingFuncWithBody(repeatElement("x = 0\n", count: 40).joined())
```

So the body should be considered to span 41 lines.

Where are you getting 42 from?

Reply:
![screen shot 2017-11-14 at 9 49 56 am](https://user-images.githubusercontent.com/19753339/32770834-38983d96-c921-11e7-9c9d-5987da12ba92.png)

I didn't touch the logic in 

```swift
    internal func exceedsLineCountExcludingCommentsAndWhitespace(_ start: Int, _ end: Int,
                                                                 _ limit: Int) -> (Bool, Int) {
        guard end - start > limit else {
            return (false, end - start)
        }

        let count = end - start - numberOfCommentAndWhitespaceOnlyLines(startLine: start, endLine: end)
        return (count > limit, count)
    }
```

with takes the endLine - startLine.

Reply:
Except that the counts _did_ change, you even had to change the tests to accommodate it.

If you look at your screenshot, the body spans from line 2 to line 42, meaning 41 lines.

Reply:
That's because the last line contains a single "}" and was computed as empty line :-)

Reply:
So the problem now isn't the `numberOfCommentAndWhitespaceOnlyLines(startLine:endLine)`. it now reports the correct number of lines that don't have code. Rules that use `exceedsLineCountExcludingCommentsAndWhitespace(_:_:_:)` have to be changed to pass the correct start line and end line of blocks.

Reply:
I wouldn't count that as the body. i.e. this function should have a body line count of 2:

```swift
func foo() { // not body
  _ = 0 // body line 1
  
  // not body, neither is above
  _ = 0 // body line 2
} // not body
```

Reply:
I agree, but not it's a different problem.

* `numberOfCommentAndWhitespaceOnlyLines`, and 
* `exceedsLineCountExcludingCommentsAndWhitespace`

report the correct number of lines. What needs to be changed now are the rules that use `exceedsLineCountExcludingCommentsAndWhitespace`.

The only reason why the rule was reporting the correct number of lines (41) was because `numberOfCommentAndWhitespaceOnlyLines` was reporting the wrong of valid lines of code. One bug was hiding the other.

Next step is to review all `_length` rules to pass the correct startLine and endLine to `exceedsLineCountExcludingCommentsAndWhitespace`exceedsLineCountExcludingCommentsAndWhitespace.

Reply:
I'll squash some of these the commits as "Fixes numberOfCommentAndWhitespaceOnlyLines" and will changed it to WIP, while the rules are not fixed.